src/components/tl/mlx5/tl_mlx5_coll.c

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5_coll.h"
+#include "mcast/tl_mcast_coll.h"
+
+static ucc_status_t ucc_tl_mlx5_task_finalize(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_mlx5_task_t *task = ucc_derived_of(coll_task, ucc_tl_mlx5_task_t);
+    tl_debug(UCC_TASK_LIB(task), "finalizing coll task %p", task);
+    UCC_TL_MLX5_PROFILE_REQUEST_FREE(task);
+    ucc_mpool_put(task);
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_mlx5_bcast_init(ucc_base_coll_args_t *coll_args,
```

Comment:
rename to ucc_tl_mlx5_bcast_mcast_init

naming scheme is ucc_<tl_name>_<collective name>_<collective algorithm name>